core/app/models/spree/promotion_handler/free_shipping.rb

```diff
@@ -20,11 +20,12 @@ def activate
       private
 
         def promotions
-          Spree::Promotion.active.where({
-            :id => Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
-            :code => nil,
-            :path => nil
-          })
+          Spree::Promotion.active.
+            joins(:promotion_codes).where({
+              :id => Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
```

Comment:
Yea we've been wanting new syntax for awhile we just don't accept commits that only change the syntax.